### PR TITLE
Root history

### PR DESCRIPTION
**Pull request by ed-**
Added in root history, and spliced it onto instances. Let's talk about how to cleanly pack it away into extensions/mysql.

### CODE REVIEW

File: reddwarf/extensions/mysql/models.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
-"""Model classes that form the core of instances functionality."""
+"""Model classes that extend the instances functionality for MySQL instances.."""
```
Comment (hub-cap) [inline reddwarf/extensions/mysql/models.py:18]: nice!

File: reddwarf/extensions/mysql/service.py [hunk 2]
```diff
@@ -74,7 +74,8 @@ def create(self, req, body, tenant_id, instance_id):
         LOG.info(_("Enabling root for instance '%s'") % instance_id)
         LOG.info(_("req : '%s'\n\n") % req)
         context = req.environ[wsgi.CONTEXT_KEY]
-        root = models.Root.create(context, instance_id)
+        user_name = req.headers.get('X-User', None)
```
Comment (hub-cap) [inline reddwarf/extensions/mysql/service.py]: We should probably put this in the middleware the builds the context

File: reddwarf/instance/service.py [hunk 3]
```diff
@@ -198,7 +198,9 @@ def show(self, req, tenant_id, id):
             LOG.error(e)
             return wsgi.Result(str(e), 404)
         # TODO(cp16net): need to set the return code correctly
-        return wsgi.Result(views.InstanceDetailView(server,
+        # Adding the root history, if it exists.
+        history = models.RootHistory.load(context=context, instance_id=id)
+        return wsgi.Result(views.InstanceDetailView(server, roothistory=history,
```
Comment (hub-cap) [inline reddwarf/instance/service.py:203]: I thought we agreed to _not_ put this in the instance service. when did this change?

File: reddwarf/extensions/mysql/service.py (hunk 2 above)
Comment (ed-) [inline reddwarf/extensions/mysql/service.py]: Yeah, you'd think we would have had that on hand already. I can do that.

File: reddwarf/instance/service.py (hunk 3 above)
Comment (ed-) [inline reddwarf/instance/service.py:203]: Short-term concession while we lacked the mgmt api in RDLI

Comment (hub-cap) [inline reddwarf/instance/service.py:203]: k

File: reddwarf/extensions/mysql/service.py (hunk 2 above)
Comment (hub-cap) [inline reddwarf/extensions/mysql/service.py]: plz do sir

**Comment by hub-cap** [comment]
+1

**Comment by TimSimpsonR** [comment]
+1

File: reddwarf/instance/models.py [hunk 4]
```diff
@@ -550,6 +552,36 @@ def __str__(self):
         return self._description
 
 
+class RootHistory(ModelBase):
```
Comment (rnirmal) [inline reddwarf/instance/models.py:555]: Hmmm... shouldn't this go in extensions/mysql/models.py. Are we punting on it for a while?

Comment (ed-) [inline reddwarf/instance/models.py:555]: No, you're very much correct. The problem's with the persisted_models function thrown to the mappers above; with the management API framework in place and a little discussion, I'd love to excise this and put it where it belongs.

Comment (rnirmal) [inline reddwarf/instance/models.py:555]: cool just making sure.

**Comment by rnirmal** [comment]
+1

**Comment by cp16net** [comment]
+1

**Comment by sacharya** [comment]
+1

**Pull request merged**